### ISSUE TITLE
LpScaler throws error in training

### ISSUE THREAD

**Issue by ganik** [microsoft/NimbusML#436]
Repro:
import numpy
from nimbusml import FileDataStream
from nimbusml.datasets import get_dataset
from nimbusml.preprocessing.normalization import LpScaler

path = get_dataset('infert').as_filepath()
data = FileDataStream.read_csv(path, sep=',')
xf = LpScaler()
features = xf.fit_transform(data)

Error: *** System.ArgumentOutOfRangeException: 'Schema mismatch for input column 'row_num': expected Expected Single or known-size vector of Single, got Single

**Comment by pieths** [comment]
Every column in the input schema is of type `Single` when it gets to the `LpScaler` transform. `LpScaler` expects a vector column of type `Single`. All the input columns will cause the error mentioned in the comment above. Here are the input column types provided to `LpScaler`:

```console
NameValueType
▶[0]{education: String}Microsoft.ML.DataViewSchema.Column
▶[1]{education: Single}Microsoft.ML.DataViewSchema.Column
▶[2]{age: Int64}Microsoft.ML.DataViewSchema.Column
▶[3]{age: Single}Microsoft.ML.DataViewSchema.Column
▶[4]{parity: Int64}Microsoft.ML.DataViewSchema.Column
▶[5]{parity: Single}Microsoft.ML.DataViewSchema.Column
▶[6]{induced: Int64}Microsoft.ML.DataViewSchema.Column
▶[7]{induced: Single}Microsoft.ML.DataViewSchema.Column
▶[8]{case: Int64}Microsoft.ML.DataViewSchema.Column
▶[9]{case: Single}Microsoft.ML.DataViewSchema.Column
▶[10]{spontaneous: Int64}Microsoft.ML.DataViewSchema.Column
▶[11]{spontaneous: Single}Microsoft.ML.DataViewSchema.Column
▶[12]{stratum: Int64}Microsoft.ML.DataViewSchema.Column
▶[13]{stratum: Single}Microsoft.ML.DataViewSchema.Column
▶[14]{pooled.stratum: Int64}Microsoft.ML.DataViewSchema.Column
▶[15]{pooled.stratum: Single}Microsoft.ML.DataViewSchema.Column
```

The examples provided with NimbusML show two different ways of using `LpScaler`.

1. Use the `collapse=True` argument to `FileDataStream.read_csv`.

    ```python
    import numpy
    from nimbusml import FileDataStream
    from nimbusml.datasets import get_dataset
    from nimbusml.preprocessing.normalization import LpScaler

    path = get_dataset('infert').as_filepath()
    data = FileDataStream.read_csv(path, sep=',',
                                   numeric_dtype=numpy.float32,
                                   collapse=True)
    xf = LpScaler(columns='age')
    features = xf.fit_transform(data)
    print(features)
    ```

    Example using a `FileDataStream` as input: https://github.com/microsoft/NimbusML/blob/master/src/python/nimbusml/examples/LpScaler.py

2. Insert a `ColumnConcatenator` before the `LpScaler` in the pipeline. 

    ```python
    import numpy
    from nimbusml import Pipeline, FileDataStream
    from nimbusml.datasets import get_dataset
    from nimbusml.preprocessing.normalization import LpScaler
    from nimbusml.preprocessing.schema import ColumnConcatenator

    path = get_dataset('infert').as_filepath()
    data = FileDataStream.read_csv(path, sep=',').to_df()

    cols = [
        'row_num', 'age', 'parity', 'induced',
        'case', 'spontaneous', 'stratum', 'pooled.stratum'
    ]
    pipeline = Pipeline([
        ColumnConcatenator() << {'Features': cols},
        LpScaler(columns='Features')
    ])

    features = pipeline.fit_transform(data)
    print(features)
    ```

    Example using a `DataFrame` as input: https://github.com/microsoft/NimbusML/blob/master/src/python/nimbusml/examples/examples_from_dataframe/LpScaler_df.py

**Comment by ganik** [comment]
This is by design. Error message will be update in ML.NET
